hdfs/util.py

```diff
@@ -29,6 +29,18 @@ def __init__(self, message, *args):
     super(HdfsError, self).__init__(message % args if args else message)
 
 
+class HdfsStandbyError(HdfsError):
```

Comment:
Let's stick to the original PR's approach of checking the status of the error inline rather than make this a first-class error since it's only used internally.

Reply:
If we revert back to HdfsError, we would need to add the information of the response to the exception, to be able to check the error later on. Which, IMHO is not so nice. Also the code is much clearer when HdfsStandbyError is used. Are you sure you want to revert back to HdfsError?

As a more general comment: I find it quite inconvenient that there is only one Exception class (HdfsError) which is raised on all kinds of errors. It is quite cumbersome to catch exceptions in code using the library and distinguish what is the cause. I would find it much nicer to have different exceptions be raised for different kinds of errors. HdfsStandbyError might be the first step in this direction ;). Although, as you pointed out, it is used only internally.

Reply:
I see your point about increasing error granularity. If we go that route, I think it'd be best done as a standalone PR; let's keep the errors consistent for now.